scripts/bert/finetune_squad.py

```diff
@@ -62,331 +62,333 @@
+        logging.info('loading vocab file from sentence piece model: %s', args.sentencepiece)
+        if dataset_name:
+            warnings.warn('Both --dataset_name and --sentencepiece are provided. '
+                          'The vocabulary will be loaded based on --sentencepiece.')
+        vocab = nlp.vocab.BERTVocab.from_sentencepiece(args.sentencepiece)
+        dataset_name = None
+    else:
+        vocab = None
+
+    pretrained = not model_parameters and not pretrained_bert_parameters and not args.sentencepiece
+    bert, vocab = nlp.model.get_model(
+        name=model_name,
+        dataset_name=dataset_name,
+        vocab=vocab,
+        pretrained=pretrained,
+        ctx=ctx,
+        use_pooler=False,
+        use_decoder=False,
+        use_classifier=False)
+
+    if args.sentencepiece:
+        tokenizer = nlp.data.BERTSPTokenizer(args.sentencepiece, vocab, lower=lower)
+    else:
+        tokenizer = nlp.data.BERTTokenizer(vocab=vocab, lower=lower)
+
+    batchify_fn = nlp.data.batchify.Tuple(
+        nlp.data.batchify.Stack(),
+        nlp.data.batchify.Pad(axis=0, pad_val=vocab[vocab.padding_token], round_to=args.round_to),
+        nlp.data.batchify.Pad(axis=0, pad_val=vocab[vocab.padding_token], round_to=args.round_to),
+        nlp.data.batchify.Stack('float32'),
+        nlp.data.batchify.Stack('float32'),
+        nlp.data.batchify.Stack('float32'))
+
+    # load symbolic model
+    deploy = args.deploy
+    model_prefix = args.model_prefix
+
+    net = BertForQA(bert=bert)
+    if model_parameters:
+        # load complete BertForQA parameters
+        nlp.utils.load_parameters(net, model_parameters, ctx=ctx, cast_dtype=True)
+    elif pretrained_bert_parameters:
+        # only load BertModel parameters
+        nlp.utils.load_parameters(bert, pretrained_bert_parameters, ctx=ctx,
+                                  ignore_extra=True, cast_dtype=True)
+        net.span_classifier.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+    elif pretrained:
+        # only load BertModel parameters
+        net.span_classifier.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+    else:
+        # no checkpoint is loaded
+        net.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+
+    net.hybridize(static_alloc=True)
+
+    loss_function = BertForQALoss()
+    loss_function.hybridize(static_alloc=True)
+
+    if deploy:
+        logging.info('load symbol file directly as SymbolBlock for model deployment')
+        net = mx.gluon.SymbolBlock.imports('{}-symbol.json'.format(args.model_prefix),
+                                           ['data0', 'data1', 'data2'],
+                                           '{}-0000.params'.format(args.model_prefix))
+        net.hybridize(static_alloc=True, static_shape=True)
+
+    # calibration config
+    only_calibration = args.only_calibration
+    num_calib_batches = args.num_calib_batches
+    quantized_dtype = args.quantized_dtype
+    calib_mode = args.calib_mode
```

Comment:
would you mind highlighting the main changes made to this script? Is it mostly changing the indentation?

Reply:
Similar as above, deploy.py imports this script for reusing the prepocessing data steps and metrics methods to check accuracy.
I moved the arguments and the parser within `if __name__ == '__main__':` for solving the conflict in the arguments expected when calling it